pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -85,46 +92,31 @@ func (o *MercuryTriggerService) RegisterTrigger(ctx context.Context, callback ch
 	o.mu.Lock()
 	defer o.mu.Unlock()
 
-	triggerID, err := o.GetTriggerID(req, wid)
+	config, err := o.ValidateConfig(req.Config)
 	if err != nil {
 		return err
 	}
 
-	// If triggerId is already registered, return an error
-	if _, ok := o.subscribers[triggerID]; ok {
-		return fmt.Errorf("triggerId %s already registered", triggerID)
-	}
-
-	cfg := subscriberConfig{}
-	err = req.Config.UnwrapTo(&cfg)
+	inputs, err := o.ValidateInputs(req.Inputs)
 	if err != nil {
 		return err
 	}
 
-	feedIDs := []mercury.FeedID{}
-	for _, feedID := range cfg.FeedIds {
-		mfid, err := mercury.NewFeedID(feedID)
-		if err != nil {
-			return err
-		}
-		feedIDs = append(feedIDs, mfid)
+	triggerID := o.GetTriggerID(inputs.TriggerID, wid)
+	// If triggerId is already registered, return an error
+	if _, ok := o.subscribers[triggerID]; ok {
+		return fmt.Errorf("triggerId %s already registered", triggerID)
 	}
 
-	if len(feedIDs) == 0 {
-		return errors.New("no feedIDs to register")
-	}
-	if cfg.MaxFrequencyMs <= 0 {
-		return errors.New("MaxFrequencyMs must be greater than 0")
-	}
-	if int64(cfg.MaxFrequencyMs)%o.tickerResolutionMs != 0 {
+	if int64(config.MaxFrequencyMs)%o.tickerResolutionMs != 0 {
```

Comment:
Sucks that we cant encode this in native JSON Schema. We can technically add extensions like https://pkg.go.dev/github.com/santhosh-tekuri/jsonschema/v5#example-package-UserDefinedFormat but IMO i dont think its worth doing for now.